source/plugin/practices.rst

```diff
@@ -61,3 +70,59 @@ Plugin Interoperability
 =======================
 
 An explanation of how to communicate with other plugins, *TBA*.
+
+
+Bad Practices
+=============
+
+These should practices should be avoided. As they lead to memory leaks (``OutOfMemoryError``), lags or inconsistencies.
+
+
+Storing References
+~~~~~~~~~~~~~~~~~~
+
+Some instances such as 
+
+* :javadoc:`DataHolder`
+* :javadoc:`TileEntity`
+* :javadoc:`CommandSource`
+* :javadoc:`Entity`
+* :javadoc:`Player`
+* :javadoc:`Subject`
+* :javadoc:`World`
+* and containers that **MIGHT** contain any of the above elements, including
+    * ``Collections``
+    * ``Maps``
+
+should **NEVER** be stored or be cached in plugins.
+
+These are the main reasons for this:
+
+* The references prevent proper garbage collection
+* The instances might no longer be valid
+
+This can easily be avoided by using the corresponding snapshots or saving the UUID of given the instances and request a
+live instance when you need it.
+
+
+IO on the main thread
+~~~~~~~~~~~~~~~~~~~~~
+
+Executing some IO operations such as loading a config/data file or checking for updates/connecting to a website takes
+much time and greatly affects the TPS on the server. Such tasks should be either done in their own threads or using the
```

Comment:
Move ``either`` ie. "Such tasks should be done either in their own thread or ..."
I'd also like to see a comma after "threads".